src/components/Reown/modals/SendTransactionModal.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect } from 'react';
+import { t } from 'ttag';
+import { useSelector, useDispatch } from 'react-redux';
+import { constants, numberUtils } from '@hathor/wallet-lib';
+import { unregisteredTokensDownloadRequested } from '../../../actions';
+import { CopyButton } from '../../CopyButton';
+
+export function SendTransactionModal({ data, firstAddress, onAccept, onReject }) {
+  const dispatch = useDispatch();
+  const { tokenMetadata, tokens: registeredTokens } = useSelector((state) => ({
+    tokenMetadata: state.tokenMetadata,
+    tokens: state.tokens,
+  }));
+
+  useEffect(() => {
+    // Collect all unregistered token UIDs
+    const unregisteredTokens = new Set();
+    
+    data?.data?.inputs?.forEach(input => {
+      const token = registeredTokens.find(t => t.uid === input.token);
```

Comment:
What happens if the token is HTR? Will it already be in the redux `tokens` array?

Reply:
Refactored this to skip HTR